langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModel.java

```diff
@@ -0,0 +1,137 @@
+
+    private final String baseUrl;
+
+    private final Double temperature;
+    private final Float topP;
+    private final String modelName;
+
+    private final String endpoint;
+    private  final Float penaltyScore;
+    private final Integer maxRetries;
+
+    private final String responseFormat;
+
+
+    @Builder
+    public QianfanChatModel(String baseUrl,
+                            String apiKey,
+                            String secretKey,
+                            Double temperature,
+                            Integer maxRetries,
+                            Float topP,
+                            String modelName,
+                            String endpoint,
+                            String responseFormat,
+                            Float penaltyScore
+                             ) {
+        if (Utils.isNullOrBlank(apiKey)||Utils.isNullOrBlank(secretKey)) {
+            throw new IllegalArgumentException(" api key and secret key must be defined. It can be generated here: https://console.bce.baidu.com/qianfan/ais/console/applicationConsole/application");
+        }
+     
+        this.modelName=modelName;
+        this.endpoint=Utils.isNullOrBlank(endpoint)? QianfanModelEnum.getEndpoint(modelName):endpoint;
+
+        if (Utils.isNullOrBlank(this.endpoint) ) {
+            throw new IllegalArgumentException("Qianfan is no such model name. You can see model name here: https://cloud.baidu.com/doc/WENXINWORKSHOP/s/Nlks5zkzu");
+        }
+
+        this.baseUrl = getOrDefault(baseUrl,  "https://aip.baidubce.com");
+
+        this.client = QianfanClient.builder().baseUrl(this.baseUrl).apiKey(apiKey).secretKey(secretKey).logStreamingResponses(true).build();
+        this.temperature = getOrDefault(temperature, 0.7);
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.topP = topP;
+        this.penaltyScore = penaltyScore;
+        this.responseFormat = responseFormat;
+    }
+
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
+
+          return  generate(messages, null,null);
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications) {
+        return generate(messages, toolSpecifications,null);
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, ToolSpecification toolSpecification) {
+        return generate(messages, singletonList(toolSpecification), toolSpecification);
+    }
+
+
+
+    private Response<AiMessage> generate(List<ChatMessage> messages,
+                                         List<ToolSpecification> toolSpecifications,
+                                         ToolSpecification toolThatMustBeExecuted
+    ) {
+
+        ChatCompletionRequest.Builder builder = ChatCompletionRequest.builder()
+                    .messages(toOpenAiMessages(messages))
+                    .temperature(temperature)
+                    .top_p(topP)
```

Comment:
please use camelCase everywhere. Gson can convert to snake_case automatically when configured:
```
Gson gson = new GsonBuilder()
    .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
    .create()
```